### PR TITLE
add tools argument to show_map

### PR DESCRIPTION
**Pull request by ekcomputer**
Passed a few base-level functions to calling functions. Added a check to prevent a confusing situation where I realized read_variables() apply_flag only affects the first variable. Added a new show_map() feature to enable an optional hover tool.

### CODE REVIEW

File: msatutil/mair_geoviews.py [hunk 1]
```diff
@@ -428,6 +432,7 @@ def read_variables(
                     v[nan_num_samples] = np.nan
                 if i == 0 and option is not None:
                     v = getattr(np, option)(v, axis=nc[var].dimensions.index(option_axis_dim))
+                    assert nc[apply_flag][:].shape == v.shape, f"Flag variable {apply_flag} has shape {nc[apply_flag][:].shape}, but first variable {var} has shape {v.shape}"
```
Comment (rocheseb) [inline msatutil/mair_geoviews.py]: this will throw an error when apply_flag is None

**Review by rocheseb — CHANGES_REQUESTED** [review]

File: msatutil/mair_geoviews.py [hunk 2]
```diff
@@ -206,6 +208,7 @@ def show_map(
         title=title,
         alpha=alpha,
         active_tools=active_tools,
+        tools=tools,
```
Comment (rocheseb) [inline msatutil/mair_geoviews.py:211]: does this still keep the default tools (pan/box_zoom/wheel_zoom/reset) when the list is empty?

File: msatutil/mair_geoviews.py (hunk 1 above)
Comment (ekcomputer) [reply msatutil/mair_geoviews.py]: Whoops, I had inserted this catch on the wrong line.

File: msatutil/mair_geoviews.py (hunk 2 above)
Comment (ekcomputer) [reply msatutil/mair_geoviews.py:211]: Yes. The active_tools options will still be turned on. There seems to be some bug where the hover tool can only be specified under the `tools` argument and the not `active_tools`, which is why I added this argument.

```python
show_map(lon, lat, var_list[0], single_panel=True, tools=['hover'])
```
image


```python
show_map(lon, lat, var_list[0], single_panel=True)
```
image

**Review by rocheseb — APPROVED** [review]

**Pull request merged**